src/iexamine.cpp

```diff
@@ -2333,6 +2333,105 @@ void iexamine::kiln_full( player &, const tripoint &examp )
     add_msg( _( "It has finished burning, yielding %d charcoal." ), result.charges );
 }
 
+void iexamine::autoclave_empty( player &p, const tripoint &examp )
+{
+    furn_id cur_autoclave_type = g->m.furn( examp );
+    furn_id next_autoclave_type = f_null;
+    if( cur_autoclave_type == furn_id( "f_autoclave" ) ) {
+        next_autoclave_type = furn_id( "f_autoclave_full" );
+    } else {
+        debugmsg( "Examined furniture has action autoclave_empty, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+
+    map_stack items = g->m.i_at( examp );
+    bool cbms = std::all_of( items.begin(), items.end(), []( const item & i ) {
+        return i.is_bionic();
+    } );
+
+    if( items.empty() ) {
+        add_msg( _( "This autoclave is empty..." ) );
+        return;
+    }
+    if( !cbms ) {
+        add_msg( m_bad,
+                 _( "You need to remove all non-CBM items from the autoclave to start the program." ) );
+        return;
+    }
+
+    auto reqs = *requirement_id( "autoclave" );
+
+    if( !reqs.can_make_with_inventory( p.crafting_inventory(), is_crafting_component ) ) {
+        popup( "%s", reqs.list_missing() );
+        return;
+    }
+
+    if( query_yn( _( "Start the autoclave?" ) ) ) {
+
+        for( const auto &e : reqs.get_components() ) {
+            p.consume_items( e, 1, is_crafting_component );
+        }
+        for( const auto &e : reqs.get_tools() ) {
+            p.consume_tools( e );
+        }
+        p.invalidate_crafting_inventory();
+        for( item it : items ) {
```

Comment:
```suggestion
        for( item &it : items ) {
```
Currently this changes a temporary item, which has no overall effect.